### ISSUE TITLE
API incompatibility with Gerrit 2.10.2

### ISSUE THREAD

**Issue by tsuna** [davido/gerrit-oauth-provider#14]
I just upgraded to Gerrit 2.10.2 and tried to switch to OAuth to avoid the impeding doom that's coming upon us on April 20th.  I went to the Google developer console, created a new project, set up the consent screen, created a new web app client ID, got the client ID and client secret, got the [pre-built jar](https://github.com/davido/gerrit-oauth-provider/releases/download/v0.1/gerrit-oauth-provider.jar) from GitHub, dropped it under the plugins directory, re-initialized Gerrit, punched in my client ID & secret, now Gerrit is coming up fine, but when I attempt to login I get:

```
[2015-04-08 00:20:41,650] WARN  org.eclipse.jetty.servlet.ServletHandler : Error for /login/%23%2Fq%2Fstatus%3Aopen
java.lang.AbstractMethodError: com.googlesource.gerrit.plugins.oauth.GoogleOAuthService.getAuthorizationUrl()Ljava/lang/String;
        at com.google.gerrit.httpd.auth.oauth.OAuthSession.login(OAuthSession.java:107)
        at com.google.gerrit.httpd.auth.oauth.OAuthWebFilter.doFilter(OAuthWebFilter.java:119)
        at com.google.gwtexpui.server.CacheControlFilter.doFilter(CacheControlFilter.java:70)
        at com.google.gerrit.httpd.RunAsFilter.doFilter(RunAsFilter.java:113)
        at com.google.gerrit.httpd.AllRequestFilter$FilterProxy$1.doFilter(AllRequestFilter.java:64)
        at com.google.gerrit.httpd.AllRequestFilter$FilterProxy.doFilter(AllRequestFilter.java:57)
        at com.google.gerrit.httpd.RequestContextFilter.doFilter(RequestContextFilter.java:75)
        at com.google.inject.servlet.ManagedFilterPipeline.dispatch(ManagedFilterPipeline.java:119)
        at com.google.inject.servlet.GuiceFilter$1.call(GuiceFilter.java:133)
        at com.google.inject.servlet.GuiceFilter$1.call(GuiceFilter.java:130)
        at com.google.inject.servlet.GuiceFilter$Context.call(GuiceFilter.java:203)
        at com.google.inject.servlet.GuiceFilter.doFilter(GuiceFilter.java:130)
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1636)
        at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:564)
        at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:219)
        at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1111)
        at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:498)
        at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:183)
        at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1045)
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:98)
        at org.eclipse.jetty.server.Server.handle(Server.java:461)
        at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:284)
        at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:244)
        at org.eclipse.jetty.io.AbstractConnection$2.run(AbstractConnection.java:534)
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:607)
        at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:536)
        at java.lang.Thread.run(Thread.java:745)
```

**Comment by tsuna** [comment]
Bleh, right after filing this issue, I found the fix from #7 (gerrit-oauth-provider@807614c50bd4d4e0797e07377a74fad61b47daea).  Need to rebuild from source.  Sigh.

Sorry for the noise.

**Comment by tsuna** [comment]
BTW, thanks for writing up the [guide to setup OAuth with Google](https://github.com/davido/gerrit-oauth-provider/wiki/Getting-Started).  It helped me navigate's Google awful UX to get things setup.  I got it to work on 2.10.2 (which required installing Buck, which required installing Ant).

Too bad Google's OAuth doesn't support naked domains, I couldn't find a way to keep our instance hosted internally at http://gerrit, which kinda blows.
